### PR TITLE
SNOW-1541942 Extend Snowpipe initialization logging at INFO level

### PR DESCRIPTION
**Pull request by sfc-gh-mbobowski**
# Overview

SNOW-1541942

Customer using Snowpipe reported random errors thrown at stage and table validations. I believe that we can log details of initialization at INFO level because it happens only on task start. One comment states that initialization happens once for 5 second with up to 120 tries but it is still not a big amount of logs. 

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [x] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java [hunk 1]
```diff
@@ -577,16 +578,17 @@ public boolean isStageCompatible(final String stageName) {
     checkConnection();
     InternalUtils.assertNotEmpty("stageName", stageName);
     if (!stageExist(stageName)) {
-      LOGGER.debug("stage {} doesn't exists", stageName);
+      LOGGER.info("stage {} doesn't exists", stageName);
```
Comment (sfc-gh-bzabek) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:581]: I'm thinking how will you find this and above log? Imagine you want to search for it in a file. When you search by "stage" or "doesn't exists" you will find many of words like this. 
Maybe we should use some convention, adding some kind of a signature for easier search? Is my concern valid, WDYT?

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java [hunk 2]
```diff
@@ -351,7 +351,7 @@ public boolean isTableCompatible(final String tableName) {
       }
       compatible = hasMeta && hasContent && allNullable;
     } catch (SQLException e) {
-      LOGGER.debug("table {} doesn't exist", tableName);
+      LOGGER.info("Table {} doesn't exist. Exception {}", tableName, e.getStackTrace());
```
Comment (sfc-gh-achyzy) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:354]: That message in logs can be interpreted as a serious situation without context. I would keep it on debug. The next one is much better for INFO.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java (hunk 1 above)
Comment (sfc-gh-achyzy) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:581]: I would keep it on debug, similar to table one, and use the next one for INFO.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java [hunk 3]
```diff
@@ -577,16 +578,17 @@ public boolean isStageCompatible(final String stageName) {
...
     if (!stageExist(stageName)) {
-      LOGGER.debug("stage {} doesn't exists", stageName);
+      LOGGER.info("stage {} doesn't exists", stageName);
       return false;
     }
     List<String> files = listStage(stageName, "");
     for (String name : files) {
       if (!FileNameUtils.verifyFileName(name)) {
-        LOGGER.debug("file name {} in stage {} is not valid", name, stageName);
+        LOGGER.info("file name {} in stage {} is not valid", name, stageName);
```
Comment (sfc-gh-achyzy) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:587]: I think we can leave it on debug.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java [hunk 4]
```diff
@@ -577,16 +578,17 @@ public boolean isStageCompatible(final String stageName) {
...
         return false;
       }
     }
+    LOGGER.info("Stage {} is valid", stageName);
```
Comment (sfc-gh-achyzy) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:591]: How about keeping it consistent with the table one? like
```suggestion
    LOGGER.info("Stage {} compatibility is {}", stageName, compatible);
```

Comment (sfc-gh-mbobowski) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java]: It required some refactor, but done

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java (hunk 3 above)
Comment (sfc-gh-mbobowski) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java]: I would like to give it a try on the INFO level. 

Customer reported an error that occasionally some files are left on a stage and it causes connector to restart. Getting file names could greatly improve our debugging capabilities and DEBUG level is unfortunately not available.

**Review by sfc-gh-achyzy — APPROVED** [review]
LGTM

**Comment by sfc-gh-mbobowski** [comment]
Close / open to restart snyk check

**Pull request merged**